Automattic-Tracks-iOS/Services/EventLoggingNetworkService.swift

```diff
@@ -0,0 +1,32 @@
+import Foundation
+
+typealias NetworkResultCallback = (Result<Data?, Error>) -> Void
+
+open class EventLoggingNetworkService {
+
+    func uploadFile(request: URLRequest, fileURL: URL, completion: @escaping NetworkResultCallback) {
+        URLSession.shared
+            .uploadTask(with: request, fromFile: fileURL, completionHandler: { data, response, error in
+
+            if let error = error {
+                completion(.failure(error))
+                return
+            }
+
+            /// The `response` should *always* be an HTTPURLResponse, but we'll be defensive anyway
+            guard let statusCode = (response as? HTTPURLResponse)?.statusCode else {
+                completion(.success(data))
```

Comment:
The reason for this is that `error` will be populated if there was an upload error, so if it's not, it seems safe to assume that the response was successful.

I also prefer to crash in these situations, so I'm very happy to refactor this. 

Changed in 4f20dc627722d58a87d15604103cd16978541198

Reply:
I was initially fine with this, but then I saw that we are checking the status code and calling the error callback if it's not in `2xx` range. That to me means either `statusCode` is absolutely necessary and we should crash, or we should call error callback if it doesn't exist. Without that bit of code, the original version of how you did this makes more sense to me. Does that make sense?

The change looks good, but there is a typo at the end of the comment. I also try to be more explicit about the comment about why we are crashing if something unexpected happens in these cases just so the next developer understands why we opted to crash. I should have mentioned this in my original comment, but anyway, here is how I would have written this comment:

```
/// The `response` should *always* be an HTTPURLResponse. To follow fail fast principle,
/// we explicitly crash here in case it's something else so we can be aware the issue and
/// address it instead of it going unnoticed.
```

Adding this type of comment in every controlled crash is a bit daunting, but I don't think this is common enough of a pattern in a8c to not do it.

As in most cases, this is just a subjective suggestion, so feel free to keep the comment as is.

Reply:
Good eyes – I adapted your suggestion for the comment in 1d9e9f886871f9b7b36fddf12eb2e551313d10d1.